src/JSInterop/Microsoft.JSInterop/test/JSRuntimeBaseTest.cs

```diff
@@ -38,6 +39,70 @@ public void DispatchesAsyncCallsWithDistinctAsyncHandles()
                 });
         }
 
+        [Fact]
+        public async Task InvokeAsync_CancelsAsyncTask_AfterDefaultTimeout()
+        {
+            // Arrange
+            var runtime = new TestJSRuntime();
+            runtime.DefaultTimeout = TimeSpan.FromSeconds(1);
+
+            // Act
+            var task = runtime.InvokeAsync<object>("test identifier 1", "arg1", 123, true);
+            await Task.Delay(TimeSpan.FromSeconds(1));
```

Comment:
Is there a risk of this being flaky? Are the continuations guaranteed to execute on the same thread and in the same order they were queued? I wouldn't have guessed that `cancellationToken.Register`'s callback necessarily ran on the same thread the call was made from.

Maybe just make the delay on L51 a bit longer than the one on L47.

Reply:
What I think you think it could happen is that task.Delay gets awaited for 1 second and between that an the call to await task the cancellation token doesn't get a chance to execute, is that assessment correct? 
It all depends on how the thread scheduler decides to execute the next task. If it decides to continue executing something else and then go and execute the continuation for task before the register callback then there's nothing we can do.

Adding more time would help reduce that chance in that place, but I'm wondering if there's something we can do to the task scheduler to force this.

In any case, I'll add more time here to mitigate that

Reply:
> What I think you think it could happen is that task.Delay gets awaited for 1 second and between that an the call to await task the cancellation token doesn't get a chance to execute, is that assessment correct?

Yes - both calls are scheduled to occur after 1 second, and may happen on different threads, so which one affects/accesses the shared state first is a bit of a gamble.

> Adding more time would help reduce that chance in that place, but I'm wondering if there's something we can do to the task scheduler to force this.

Instead of waiting for 1 second, we could wait for `task` itself to complete, then assert that it failed with a `TaskCancellationException`.

Reply:
Yeah, you are right, the task.delay there does nothing. This is why I shouldn't be writing tests at night 😄